File: Codeforces/Div4/784/Triple/main.py
# https://codeforces.com/contest/1669/problem/B

import io, os, sys

input = io.BytesIO(os.read(0, os.fstat(0).st_size)).readline

def readline():
    return input().decode()

def readint(): 
    return int(readline())

def readintlist():
    return list(map(int, readline().split()))

t = readint()

while(t):
    dic = {}
    result = -1
    n = readint()
    a = readintlist()

    for ai in a:
        if dic.get(ai):
            dic[ai] += 1
        else:
            dic[ai] = 1
        
        if dic[ai] >= 3:
            result = ai

    sys.stdout.write(str(result) + '\n')
    t -= 1

File: Codeforces/Div4/784/Triple/main.cpp
// https://codeforces.com/contest/1669/problem/B

#include<bits/stdc++.h>

using namespace std;

int t, n, ai, result;

int main () {
    cin.tie(0);
    ios_base::sync_with_stdio(0);

    cin >> t;

    while(t--){
        cin >> n;

        int a[n+2];

        result = -1;
        memset(a, 0, sizeof a);
        
        for (int i=0; i<n; i++) {
            cin >> ai;
            a[ai]++;

            if (a[ai] >= 3)
                result = ai;
        }

        cout << result << endl;
    }

    return 0;
}
